### PR TITLE
Line limits assertion fix

### PR DESCRIPTION
**Pull request by nkorinek**
Changed how the limits of a plotted line are checked to make it more generous for small mistakes.
addresses: https://github.com/earthlab/matplotcheck/issues/235

### CODE REVIEW

**Comment by lwasser** [comment]
hey @nkorinek looks like we have some conflicts now. also i'm seeing build errors. is pillow installed in your dev envt? it seems like it's missing from the travis errors. i'm curious whether we should add it as a dev requirement?

**Comment by lwasser** [comment]
actually maybe it's the same m2r issue??

**Comment by nkorinek** [comment]
@lwasser I've updated it to include the m2r fix. Hopefully that fixes the issues you were seeing

**Comment by lwasser** [comment]
great. thank you @nkorinek !!

**Comment by lwasser** [comment]
@nkorinek are you still working on this guy? is it ready to merge if CI passes or is it a WIP?

**Comment by nkorinek** [comment]
@lwasser fixing a bug with this right now, once I push up the changes it should be good to go

**Comment by nkorinek** [comment]
@lwasser did more testing on this and it should be good to merge once that issue with CI is fixed

**Comment by lwasser** [comment]
thank you @nkorinek i'm going to give filipe a day or two to respond... and if he doesn't i'll come up with plan b :) but i suspect he will give me some suggestions regarding things to try!!

File: matplotcheck/base.py
Comment (lwasser) [inline matplotcheck/base.py:1109]: @nkorinek can you please add a comment here that explains how this fixes this issue. i understand the issue at a high level. so what if the data have a few extreme edge cases outlier points. and the regression line only fits to 3/4 of the data or 1/2 of the data. will this fail? i think i just need a bit more clarification 

did we discuss making the data limits of a line optional? that could be something else to consider OR creating a regression line and provided it to the method. i just want to talk this through a bit more so i understand and i want the code to have a comment for the same reason. thank you again!

**Review by lwasser — CHANGES_REQUESTED** [review]
i have a question about how the data lims are tested here! ALso we have change log issues now. as always :)

**Comment by lwasser** [comment]
ok this guy still needs a bit of love @nkorinek before i can merge it. see my questions above.  i know i need to merge this prior to merging the lines vignette so let's focus on this guy first. 
update from master and then ofcourse the change log needs an update (md vs rst)

**Comment by nkorinek** [comment]
@lwasser oh sorry! I added a comment in the code, but I'll clarify here as well and see if this is what we want to do. 

> so what if the data have a few extreme edge cases outlier points. and the regression line only fits to 3/4 of the data or 1/2 of the data. will this fail? 

This will not fail in this case. In fact most of the time the line does not cover the entirety of the limits, and will be within the bounds but not close to the x and y limits. It seems to do this on lines that should pass normally, however, so maybe that's why this was done this way before? Originally I had this JUST check if the limits were close to the expected limits, but that broke a lot of the tests, so if it checks that it's either within the limits or very close to the limits that's how I got around that issue. Checking that the line was within the limits is how the check was run before. We could change this check if we wanted to!  

> did we discuss making the data limits of a line optional? that could be something else to consider

I don't remember discussing this, but we may have discussed it and I just forgot. Either way, it would be fairly simple to add, but I'm not sure why you would turn that check off. The only time I would turn that check off is if the check was failing and it wasn't meant to fail, but hopefully this PR fixes that. Let me know what you think!

**Comment by lwasser** [comment]
let's just add a few tests to ensure this works in all combinations of points + outlier points

**Comment by nkorinek** [comment]
@lwasser this can be closed due to this pr: #274

**Pull request closed without merging**
